app/src/main/java/de/westnordost/streetcomplete/quests/access_point_ref/AddAccessPointRefForm.kt

```diff
@@ -0,0 +1,53 @@
+package de.westnordost.streetcomplete.quests.access_point_ref
+
+import android.os.Bundle
+import android.view.View
+import androidx.appcompat.app.AlertDialog
+import androidx.core.widget.doAfterTextChanged
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.osm.mapdata.Node
+import de.westnordost.streetcomplete.databinding.QuestRefBinding
+import de.westnordost.streetcomplete.quests.AbstractOsmQuestForm
+import de.westnordost.streetcomplete.quests.AnswerItem
+
+class AddAccessPointRefForm : AbstractOsmQuestForm<AccessPointRefAnswer>() {
+
+    override val contentLayoutResId = R.layout.quest_ref
+    private val binding by contentViewBinding(QuestRefBinding::bind)
+
+    override val otherAnswers get() = listOfNotNull(
+        AnswerItem(R.string.quest_ref_answer_noRef) { confirmNoRef() },
+        createMarkAsAssemblyPointAnswer()
+    )
+
+    private val ref get() = binding.refInput.text?.toString().orEmpty().trim()
```

Comment:
Rather use
```suggestion
    private val ref get() = binding.refInput.nonBlankTextOrNull
```
and correct other places accordingly

Reply:
How it should be handled in `applyAnswer(AccessPointRef(ref))`? 

`applyAnswer(AccessPointRef(ref!!))`

?

`!!` should be avoided, but admittedly it is not worse than current situation where bug could in theory let empty string end there,

Reply:
yes, !!

Reply:
BTW, I applied it to the current code - including one which was a base for this quest

in https://github.com/streetcomplete/StreetComplete/commit/0aaffb13d359714036271a3c212d6c65772321c4